contracts/transfer-types/src/lib.rs

```diff
@@ -145,6 +145,18 @@ pub struct Stct {
     pub proof: Proof,
 }
 
+/// Send value to a contract transparently.
+#[derive(Debug, Clone, Archive, Deserialize, Serialize)]
+#[archive_attr(derive(CheckBytes))]
+pub struct Stct2 {
+    /// Module to send the value to.
+    pub address: ModuleId,
+    /// The value to send to the contract.
+    pub value: u64,
+    /// Proof of the `STCT` circuit.
+    pub proof: Vec<u8>,
+}
```

Comment:
Looking at the difference between `Stct` and `Stct2` I can understand why the structure was introduced, but by changing the signature of the stake contract functions to take a `Proof` instead of a `Vec<u8>` (as it should be), we avoid this entirely.